### PR TITLE
Add type hints for lib/corpus.py

### PR DESCRIPTION
**Pull request by JackTemaki**
While writing a job I was again annoyed that there are no type hints or the corpus lib. So I added them now.

### CODE REVIEW

**Review by christophmluscher — APPROVED** [review]

**Review by mattiadg — COMMENTED** [review]
Good, only suggesting minor changes

File: lib/corpus.py [hunk 1]
```diff
@@ -31,7 +41,7 @@ class CorpusParser(sax.handler.ContentHandler):
     is currently beeing read.
     """
 
-    def __init__(self, corpus, path):
+    def __init__(self, corpus: Corpus, path: str):
```
[the remark this answers is not in the document]
Comment (JackTemaki) [reply lib/corpus.py:44]: I do not feel confident enough with the pathlib to just change the code in a few minutes now, but we can keep this as issue for later.

File: lib/corpus.py [hunk 2]
```diff
@@ -1,25 +1,35 @@
...
 
 
+FilterFunction = Callable[["Corpus", "Recording", "Segment"], bool]
+
+
 class NamedEntity:
     def __init__(self):
         super().__init__()
-        self.name = None
+        self.name = None  # type: Optional[str]
```
[the remark this answers is not in the document]
Comment (JackTemaki) [reply lib/corpus.py:25]: I am still way too used to old code syntax...

Sure will change.

File: lib/corpus.py [hunk 3]
```diff
@@ -330,16 +341,16 @@ def __init__(self):
...
 
-    def fullname(self):
+    def fullname(self) -> str:
         return self.recording.fullname() + "/" + self.name
 
-    def speaker(self):
+    def speaker(self) -> Speaker:
         return self.recording.speaker(self.speaker_name)
 
     def dump(self, out, indentation=""):
```
Comment (michelwi) [inline lib/corpus.py:356]: you missed a few functions 👋

Comment (JackTemaki) [reply lib/corpus.py:356]: I did not miss them, I was too lazy to look up the type for out :) 

But okay, as I got caught I will search for it.

Comment (michelwi) [reply lib/corpus.py:356]: `typing.TextIO` ?
https://docs.python.org/3/library/typing.html#typing.IO

Comment (JackTemaki) [reply lib/corpus.py:356]: It should be all `TextIO` right?

**Review by michelwi — APPROVED** [review]

**Pull request merged**